tiledb/sm/c_api/tiledb.cc

```diff
@@ -3274,6 +3274,129 @@ int32_t tiledb_array_encryption_type(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_put_metadata(
+    tiledb_ctx_t* ctx,
+    tiledb_array_t* array,
+    const char* key,
+    tiledb_datatype_t value_type,
+    uint32_t value_num,
+    const void* value) {
+  if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, array) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Put metadata
+  if (SAVE_ERROR_CATCH(
+          ctx,
+          array->array_->put_metadata(
+              key,
+              static_cast<tiledb::sm::Datatype>(value_type),
+              value_num,
+              value)))
+    return TILEDB_ERR;
+
+  return TILEDB_OK;
+}
+
+int32_t tiledb_array_delete_metadata(
+    tiledb_ctx_t* ctx, tiledb_array_t* array, const char* key) {
+  if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, array) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Put metadata
+  if (SAVE_ERROR_CATCH(ctx, array->array_->delete_metadata(key)))
+    return TILEDB_ERR;
+
+  return TILEDB_OK;
+}
+
+int32_t tiledb_array_get_metadata(
+    tiledb_ctx_t* ctx,
+    tiledb_array_t* array,
+    const char* key,
+    tiledb_datatype_t* value_type,
+    uint32_t* value_num,
```

Comment:
I wonder if the `get()` functions should be returning const pointers.

Reply:
Notice that `value` is passed as a constant pointer because we just point to the value stored in the metadata object (i.e., we don't copy). In contrast, `value_type` and `value_num`, being fixed-sized small objects, are just copied instead of passing the pointer to the values stored in the metadata object. I think either way I am fine.

Reply:
Oh I see, copying seems fine. I was misreading the function signature here.